src/test/java/io/strimzi/kafka/metrics/KafkaMetricsCollectorTest.java

```diff
@@ -51,35 +44,67 @@ public void testCollect() {
         assertTrue(metrics.isEmpty());
 
-        // Adding a non-numeric metric does nothing
-        collector.addMetric(buildNonNumericMetric("name2", "group"));
-        metrics = collector.collect();
-        assertTrue(metrics.isEmpty());
-
         // Adding a metric that matches the allowlist
         collector.addMetric(buildMetric("name", "group", 1.0));
         metrics = collector.collect();
         assertEquals(1, metrics.size());
-        assertEquals("kafka_server_group_name", metrics.get(0).name);
-        assertEquals(1, metrics.get(0).samples.size());
-        assertEquals(1.0, metrics.get(0).samples.get(0).value, 0.1);
-        assertEquals(new ArrayList<>(labels.keySet()), metrics.get(0).samples.get(0).labelNames);
-        assertEquals(new ArrayList<>(labels.values()), metrics.get(0).samples.get(0).labelValues);
+
+        Collector.MetricFamilySamples metricFamilySamples = metrics.get(0);
+        assertMetricFamilySample(metricFamilySamples, "kafka_server_group_name", 1.0, labels);
 
         // Adding the same metric updates its value
         collector.addMetric(buildMetric("name", "group", 3.0));
         metrics = collector.collect();
         assertEquals(1, metrics.size());
-        assertEquals("kafka_server_group_name", metrics.get(0).name);
-        assertEquals(1, metrics.get(0).samples.size());
-        assertEquals(3.0, metrics.get(0).samples.get(0).value, 0.1);
+
+        Collector.MetricFamilySamples updatedMetrics = metrics.get(0);
+        assertMetricFamilySample(updatedMetrics, "kafka_server_group_name", 3.0, labels);
 
         // Removing the metric
         collector.removeMetric(buildMetric("name", "group", 4.0));
         metrics = collector.collect();
         assertTrue(metrics.isEmpty());
     }
 
+    @Test
+    public void testCollectNonNumericMetric() {
+        Map<String, String> props = new HashMap<>();
+        props.put(PrometheusMetricsReporterConfig.ALLOWLIST_CONFIG, "kafka_server_group_name.*");
+        PrometheusMetricsReporterConfig config = new PrometheusMetricsReporterConfig(props);
+        KafkaMetricsCollector collector = new KafkaMetricsCollector(config);
+        collector.setPrefix("kafka.server");
+
+        List<Collector.MetricFamilySamples> metrics = collector.collect();
+        assertTrue(metrics.isEmpty());
+
+        // Adding a non-numeric metric converted
+        String nonNumericValue = "myValue";
+        KafkaMetric nonNumericMetric = buildNonNumericMetric("name", "group", nonNumericValue);
+        collector.addMetric(nonNumericMetric);
+        metrics = collector.collect();
+
+        Map<String, String> expectedLabels = new LinkedHashMap<>(labels);
+        expectedLabels.put("name", nonNumericValue);
+        assertEquals(1, metrics.size());
+
+        Collector.MetricFamilySamples metricFamilySamples = metrics.get(0);
+
+        assertEquals("kafka_server_group_name", metricFamilySamples.name);
+        assertEquals(1, metricFamilySamples.samples.size());
```

Comment:
These 2 assertions are checked again in `assertMetricFamilySample()`. Why are we checking them twice?

Reply:
Duplicates removed

Reply:
I think it would be more idiomatic to either check the size in `assertMetricFamilySample()`, or pass the unique sample (`metricFamilySamples.samples.get(0)`) to `assertMetricFamilySample()`.

It's weird to check the sample size is one and still pass the list of samples to `assertMetricFamilySample()`.

Same in `YammerMetricsCollectorTest`.